service/src/main/java/aero/minova/cas/MultipleLdapDomainsAuthenticationProvider.java

```diff
@@ -0,0 +1,66 @@
+package aero.minova.cas;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.ldap.authentication.ad.ActiveDirectoryLdapAuthenticationProvider;
+import org.springframework.security.ldap.userdetails.UserDetailsContextMapper;
+
+public class MultipleLdapDomainsAuthenticationProvider implements AuthenticationProvider {
+
+	List<ActiveDirectoryLdapAuthenticationProvider> providers = new ArrayList<>();
+
+	public MultipleLdapDomainsAuthenticationProvider(String domains, String ldapServerAddresses, UserDetailsContextMapper userDetailsContextMapper) {
+
+		List<String> domainList = Arrays.asList(domains.split(SecurityConfig.MULTIPLE_LDAP_CONFIGURATIONS_SEPERATOR));
+		List<String> addressList = Arrays.asList(ldapServerAddresses.split(SecurityConfig.MULTIPLE_LDAP_CONFIGURATIONS_SEPERATOR));
+
+		for (int i = 0; i < domainList.size(); i++) {
+
+			String address = addressList.size() == 1 ? addressList.get(0) : addressList.get(i);
```

Comment:
Wozu ist das if da?

Reply:
Wie auch in der Doku geschrieben kann man nur eine Adresse für alle Domänen nutzen, ich schreib noch einen Kommentar an die Stelle im Code.

Reply:
Entweder genau einen Server für alle Domänen, oder man muss halt für jede Domäne auch den Server konfigurieren.

So habe ich es auch in die Doku geschrieben, und auch noch einen Check eingebaut, dass die Anzahl passen muss.